client/src/pages/mycourses/MyCourses.jsx

```diff
@@ -1,23 +1,45 @@
+import { useEffect, useState } from 'react';
 import MyCourseCard from '../../components/mycoursescard/MyCourseCard.jsx';
-import './MyCourses.css'
+import './MyCourses.css';
+import CourseService from '../../services/CourseService';
+import { useNavigate } from 'react-router-dom';
 
 export default function MyCourses() {
-  return(
+  const navigate = useNavigate();
+  const [courses, setCourses] = useState([]);
+
+  useEffect(() => {
+    const fetchEnrolledCourses = async () => {
+      try {
+        const coursesData = await CourseService.getEnrolledCourses(1);
+        console.log(coursesData);
+        setCourses(coursesData);
+      } catch (error) {
+        alert('ERROR: Failed to fetch enrolled courses');
+      }
+    };
+
+    fetchEnrolledCourses();
+  }, []);
+
+  return (
     <div className="my-courses">
       <div className="enrolled-courses-title">Enrolled Courses</div>
 
       <div className="mycourses-list-container">
         <div className="mycourses-list">
-          <MyCourseCard courseTitle="Course 1" imageUrl="" instructorName="Instructor 1" progress={100}/>
-          <MyCourseCard courseTitle="Course 2" imageUrl="" instructorName="Instructor 2" progress={75}/>
-          <MyCourseCard courseTitle="Course 3" imageUrl="" instructorName="Instructor 3" progress={50}/>
-          <MyCourseCard courseTitle="Course 4" imageUrl="" instructorName="Instructor 4" progress={30}/>
-          <MyCourseCard courseTitle="Course 5" imageUrl="" instructorName="Instructor 5" progress={0}/>
-          <MyCourseCard courseTitle="Course 6" imageUrl="" instructorName="Instructor 6" progress={10}/>
-
+          {courses.map(c => (
+            <MyCourseCard
+              key={c.enrolledCourse.id}
+              courseTitle={c.enrolledCourse.title}
+              imageUrl=""
+              instructorName={c.enrolledCourse.description}
+              progress={c.progress}
+              handleClick={() => navigate(`/courseviewer/${c.id}`)}
+            />
+          ))}
```

Comment:
love me some mapping :)

Reply:
Absolute favorite part about coding. Functional programming FTW. 

I used to not be a huge fan of it and things like [this](https://github.com/nickerick/skill-magnet/blob/288668759b96ecaf8c01f130985f6133b32a8818/server/src/main/java/com/skillmagnet/Lesson/LessonController.java#L69) were the most I approved of. But at my current internship they have some pretty disgusting mappings that go on where it'll be like 15 methods chained together. I've been stockholmed syndromed into loving it now. Also where Optionals are super useful and make much more sense than how we're using them lol

If you go down to the 'Less Boilerplate' section of this [article](https://belief-driven-design.com/functional-programming-with-java-streams-190eda591a5/) you can be gateway drugged into the functional programming cult.

Reply:
> I've been stockholmed syndromed into loving it now

lmao

> Also where Optionals are super useful and make much more sense than how we're using them lol

I definitely see the potential.

```java
List<String> result = 
    albums.stream()
          .filter(album -> album.getYear == 1999)
          .filter(album -> album.getGenre() == Genre.ALTERNATIVE)
          .limit(5)
          .map(Album::getName)
          .sorted()
          .collect(Collectors.toList());
```
This is beautiful. Doesn't even look like java to me, if you know what I mean lol

Reply:
Life is good when all your logic looks like this. It's so alien at first but once you get used to it you realize how much faster it is to write code and also figure out what something is doing